Sources/ClientRuntime/Endpoints/EndpointsRequestContextProviding.swift

```diff
@@ -0,0 +1,10 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public protocol EndpointsRequestContextProviding {
+    var context: EndpointsRequestContext { get throws }
+}
```

Comment:
This protocol enables endpoint params to provide their own context.  Context generation used to be performed by the endpoint resolver.